app/controllers/profiles_controller.rb

```diff
@@ -1,16 +1,13 @@
-
 class ProfilesController < ApplicationController
-  include LolConnections
   before_action :require_user, only: [:show]
   before_action :check_or_set_user, except: [:index]
-  before_action :verify_summoner_name_and_stats, only: [:create]
 
   def index
   end
 
   def show
     @user ||= current_user
-    featured_stats_service = FeaturedStatsService.new(@user)
+    featured_stats_service = FeaturedStatsService.new(@user.name)
```

Comment:
I think the issue was because the username I was testing with was no longer active. I tried a new username and I was able to login. What I found was that after pulling up a signed in user's friend's list, if you click on the "View Profile" button for a listed friend, it doesn't redirect to the friend's page. Instead, it redirects to the signed in user's profile page.
My observation is that this is due to the change made in the show method of Profiles controller. Before, we simply gave show an id and it will show that id. Currently, show doesn't use an id, it just shows the logged in user's profile. I looked at the console and saw that even though the id was passed in, show was using the logged in user's id instead.
I don't think this problem arose because of your recent change. I think it came from an earlier refactor (search-refactor branch?). Could you test my observation before and after search refactor and let me know what you get?